engine/src/main/java/org/pentaho/di/trans/step/BaseStep.java

```diff
@@ -4410,5 +4412,38 @@ private class DefaultRowHandler implements RowHandler {
     }
 
   }
+
+  @Override
+  public JSONObject doAction( String fieldName, StepMetaInterface stepMetaInterface, TransMeta transMeta,
+                               Trans trans, Map<String, String> queryParams ) {
+    JSONObject response = new JSONObject();
+    try {
+      Method actionMethod = this.getClass().getDeclaredMethod( fieldName + "Action", Map.class );
```

Comment:
We could replace reflection with a call to a method that takes fieldName as its first argument. That method could then be overridden by subclasses to just map fieldName to the appropriate internal method, still keeping the rest of the boilerplate in the base class.
I think this would make it easier for someone reading the code to find where those Action methods are being called.

From what I could gather these seem to be related to configuration editing, and therefore Step classes may not be the best place for them, as noted in another comment.

Also noticed that BaseStep and DefaultRowHandler share the exact same boilerplate, which could probably be moved to a static utility method.